src/testshade/testshade.cpp

```diff
@@ -390,15 +390,7 @@ action_groupspec (int argc, const char *argv[])
     if (OIIO::Filesystem::exists (groupspec)) {
         // If it names a file, use the contents of the file as the group
         // specification.
-        std::ifstream in;
-        OIIO::Filesystem::open (in, groupspec);
-        groupspec.clear();
-        while (in.good()) {
-            std::string line;
-            std::getline (in, line);
-            groupspec += line + "\n";
-        }
-        in.close ();
+        OIIO::Filesystem::read_text_file (groupspec, groupspec);
```

Comment:
It does lead to an interesting rule of thumb though: if you write a function that takes a mutable string and a string_view, it is dangerous to use the string_view after you mutate the string, just in case somebody passed a view of that same string.

Though, this is a similar kind of weird you could get if you passed in one const std::string&, and another std::string& (meant to be modified). You won't have memory issues, but you can have "oops, the string I thought was read-only changed half way through the function."